### PR TITLE
🪟 🐛 Fix duplicate createUser calls on signup

### PR DESCRIPTION
**Pull request by joeykmh**
## What
`onAfterAuth()` was being called multiple times. This in turn was causing multiple calls to `web_backend/users/create` with the same user information, which caused many 409s (conflicting user names) and spamming our oncall with failed requests.

## How
Reverts the change introduced in https://github.com/airbytehq/airbyte/pull/21175

## Considerations
The auth service is missing tests, which could help avoid this. The component is complex and depends on many other modules, so testing this is out of scope for this PR.

### CODE REVIEW

File: airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx [hunk 1]
```diff
@@ -147,26 +147,34 @@ export const AuthenticationProvider: React.FC<React.PropsWithChildren<unknown>>
   const stateRef = useRef(state);
   stateRef.current = state;
 
-  useEffectOnce(() => {
-    return auth.onAuthStateChanged(async (currentUser) => {
-      // We want to run this effect only once on initial page opening
+  const onAuthStateChange = useCallback(
```
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx:150]: Extracted since this logic is used in two places

**Comment by josephkmh** [comment]
@tealjulia FYI

File: airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx [hunk 2]
```diff
@@ -147,26 +147,34 @@ export const AuthenticationProvider: React.FC<React.PropsWithChildren<unknown>>
...
 
+  // Check auth state on window focus, in case the user logged out in a different tab
   useEffect(() => {
     const onFocus = async () => {
       return auth.onAuthStateChanged(async (currentUser) => {
         if (!currentUser) {
           loggedOut();
         } else {
-          await onAfterAuth(currentUser);
+          onAuthStateChange(currentUser);
```
Comment (yeomann) [inline airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx:177]: with this new PR changes, async is redundant at `const onFocus = async` and `return auth.onAuthStateChanged(async`

Comment (josephkmh) [reply airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx:177]: Thanks for catching that, indeed there are several unnecessary `async` keywords, will clean those up 🙂

Comment (yeomann) [reply airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx:177]: no worries, i just came across the repo today and was checking some pull requests.

**Review by flash1293 — APPROVED** [review]
LGTM; as discussed in Slack this is a simple revert of https://github.com/airbytehq/airbyte/pull/21175 - a new issue will be created to track the new proper fix of that

**Pull request merged**